iis-workflows/iis-workflows-citationmatching-direct/src/main/java/eu/dnetlib/iis/workflows/citationmatching/direct/CitationMatchingDirectJob.java

```diff
@@ -0,0 +1,133 @@
+import eu.dnetlib.iis.workflows.citationmatching.direct.converters.DirectCitationToCitationConverter;
+import eu.dnetlib.iis.workflows.citationmatching.direct.converters.DocumentToDirectCitationMetadataConverter;
+import eu.dnetlib.iis.workflows.citationmatching.direct.model.IdWithPosition;
+import eu.dnetlib.iis.workflows.citationmatching.direct.service.ExternalIdCitationMatcher;
+
+
+public class CitationMatchingDirectJob {
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    public static void main(String[] args) throws IOException {
+        
+        CitationMatchingDirectJobParameters params = new CitationMatchingDirectJobParameters();
+        JCommander jcommander = new JCommander(params);
+        jcommander.parse(args);
+        
+        
+        SparkConf conf = new SparkConf();
+        conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
+        conf.set("spark.kryo.registrator", "eu.dnetlib.iis.core.spark.AvroCompatibleKryoRegistrator");
+        
+        
+        
+        try (JavaSparkContext sc = new JavaSparkContext(conf)) {
+            
+            JavaRDD<ExtractedDocumentMetadataMergedWithOriginal> documentsMetadata = SparkAvroLoader.loadJavaRDD(sc, params.inputAvroPath, ExtractedDocumentMetadataMergedWithOriginal.class);
+            
+            
+            DocumentToDirectCitationMetadataConverter documentMetadataConverter = new DocumentToDirectCitationMetadataConverter();
+            JavaRDD<DocumentMetadata> simplifiedDocumentsMetadata = documentsMetadata.map(metadata -> documentMetadataConverter.convert(metadata));
+//            simplifiedDocumentsMetadata = simplifiedDocumentsMetadata.cache(); // FIXME: https://github.com/openaire/iis/issues/128
+            
+            JavaRDD<Citation> directDoiCitations = matchDoiCitations(simplifiedDocumentsMetadata);
+            
+            JavaRDD<Citation> directPmidCitations = matchPmidCitations(simplifiedDocumentsMetadata);
+            
+            JavaRDD<Citation> directCitations = mergeCitations(directDoiCitations, directPmidCitations);
+            
+            
+            
+            DirectCitationToCitationConverter directCitationToCitationConverter = new DirectCitationToCitationConverter();
+            JavaRDD<eu.dnetlib.iis.common.citations.schemas.Citation> citations = 
+                    directCitations.map(directCitation -> directCitationToCitationConverter.convert(directCitation));
+            
+            
+            SparkAvroSaver.saveJavaRDD(citations, eu.dnetlib.iis.common.citations.schemas.Citation.SCHEMA$, params.outputAvroPath);
+        }
+        
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    
+    private static JavaRDD<Citation> matchDoiCitations(JavaRDD<DocumentMetadata> directCitationMetadata) {
+        ExternalIdCitationMatcher externalIdCitationMatcher = new ExternalIdCitationMatcher();
```

Comment:
I would move it to the static section of this class and refer to it indirectly in main method. Thanks to that it will be more similar to standard solutions in which we have the section of service definitions and then, in methods, their invocations.